workflow-templates/im-test-selenium-manual.yml

```diff
@@ -0,0 +1,207 @@
+  cache:
+    runs-on: im-linux
+    if: ${{ inputs.enable-cache }}
+
+    steps:
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
+      - name: Calculate Ref to Checkout
+        id: calculate-ref
+        shell: bash
+        run: |
+          GITHUB_REF="${{ env.DISPATCH_REF }}"
+          echo "GITHUB_REF=$GITHUB_REF" >> $GITHUB_ENV
+          echo "GITHUB_REF=$GITHUB_REF" >> $GITHUB_OUTPUT
+
+      - name: Annotate Inputs
+        shell: bash
+        run: |
+          echo $'
+          | Inputs                  | Value                                                                                     |
+          | ----------------------- | ----------------------------------------------------------------------------------------- |
+          | Environment             | `${{ inputs.environment }}`                                                               |
+          | Branch/Tag              | `${{ env.GITHUB_REF }}`                                                                   |
+          | Workflow Branch/Tag     | `${{ github.ref_name }}` - SHA: `${{ github.sha }}`                                       |
+          | Browser                 | `${{ inputs.browser }}`                                                                   |
+          | Test Filter             | `${{ inputs.test-filter != 0 && inputs.test-filter || 'NA' }}`                            |' >> $GITHUB_STEP_SUMMARY
+
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.GITHUB_REF }}
+
+      - name: Set Cache Keys
+        shell: bash
+        run: echo "DOTNET_CACHE_KEY=dotnet_packages-${{ hashFiles('**/*.csproj') }}" >> $GITHUB_ENV
+
+      - name: Authenticate with GitHub Packages
+        uses: im-open/authenticate-with-gh-package-registries@v1.1
+        with:
+          github-token: ${{ secrets.READ_PKG_TOKEN }}
+                
+      - name: Check for a .NET cache
+        id: has-cache
+        uses: actions/cache@v4
+        with:
+          lookup-only: true
+          key: ${{ env.DOTNET_CACHE_KEY }}
+          path: |
+            ~/.nuget/packages
+
+      - name: Restore Dependencies
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        shell: bash
+        run: dotnet restore ${{ env.SELENIUM_TEST_DIR }}
+
+      - name: Save cache for .NET packages
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          key: ${{ env.DOTNET_CACHE_KEY }}
+          path: |
+            ~/.nuget/packages
+
+  build-and-test:
+    runs-on: im-linux
+    needs: cache
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.DISPATCH_REF }}
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
```

Comment:
You need a step to download/restore the nuget cache.

Reply:
good catch, added.

Reply:
Thanks for catching that. I've added it.